### PR TITLE
Add crash immediate with syscall test app

### PR DESCRIPTION
**Pull request by bradjc**
This is a very small test app for testing what happens if an app crashes on its first instruction.

### CODE REVIEW

File: examples/tests/crash_immediately_syscall/main.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+            void* mem_start __attribute__((unused)),
+            void* memory_len __attribute__((unused)),
+            void* app_heap_break __attribute__((unused))) {
+  // invalid syscall
+  __asm__ volatile (
+      "svc 0xbd       \n"
+      :
+      :
+      : "memory", "r0", "r1", "r2", "r3", "r12", "lr"
+      );
```
Comment (alistair23) [inline examples/tests/crash_immediately_syscall/main.c:16]: This will fail to build fro RISC-V

**Review by alistair23 — CHANGES_REQUESTED** [review]

**Review by hudson-ayers — COMMENTED** [review]
Looks good, but agree with alistair that adding risc-v support is probably important

**Comment by ppannuto** [comment]
Pushed a commit to mark this as a naked function (otherwise gcc will emit a prologue instruction that we don't really want for this). Also pushed what I think is the right thing for riscv, but something is broken on this machine with that toolchain that is beyond tonight's patience to debug.

**Comment by hudson-ayers** [comment]
amended your commit with a format run so we can just see if travis can build the app for risc-v

**Comment by ppannuto** [comment]
Oops. Thanks. The riscv toolchain is also fixed now on my machine (just needed to rebuild for reasons unknown, which became an overnight chore for ye olde laptop), and I can confirm it builds there happily as well.

**Review by hudson-ayers — APPROVED** [review]

**Review by alistair23 — APPROVED** [review]

File: examples/tests/crash_immediately_syscall/main.c [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#if defined(__thumb__)
+  __asm__ volatile (
+    "svc 0xbd       \n"
+    :
+    :
+    : "memory", "r0", "r1", "r2", "r3", "r12", "lr"
+    );
+#elif defined(__riscv)
+  __asm__ volatile (
+    "li    a4, 0xbd\n"
```
Comment (bradjc) [inline examples/tests/crash_immediately_syscall/main.c:21]:
```suggestion
    "li    a0, 0xbd\n"
```
?

Comment (lschuermann) [reply examples/tests/crash_immediately_syscall/main.c:21]: `a4` is the syscall argument register on RISC-V now: https://github.com/tock/tock/blob/master/doc/reference/trd104-syscalls.md#31-registers

File: examples/tests/crash_immediately_syscall/main.c [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// Do nothing. yield() immediately.
```
Comment (lschuermann) [inline examples/tests/crash_immediately_syscall/main.c:1]: I think this comment is slightly misleading. :)

**Pull request merged**